tests/components/stream/test_worker.py

```diff
@@ -107,8 +117,8 @@ def __init__(self, video_stream, audio_stream):
         self.packets = PacketSequence(0)
 
         class FakePyAvStreams:
-            video = video_stream
-            audio = audio_stream
+            video = [video_stream] if video_stream else []
+            audio = [audio_stream] if audio_stream else []
```

Comment:
If this should be a fake pyavclass, I'm a bit surprised by this change. Was it completely invalid before?

Reply:
It worked fine before. The only reason for the change is because of L174:175 / L184:185 below. Previously the video_stream itself was defined as a list which seemed odd, so this just changes the definition below and adds the list here.

Reply:
Yes, but I assume this tries to match the real pyav structures. And both these can't be correct.

Reply:
Sorry I don't quite follow..by both do you mean the previous way vs the PR way?

Reply:
Yes i mean before the pull and after. One of them doesn't match real pyav structures. So I'm trying to deduce witch.

Reply:
There's no difference with these two lines when taken along with the changes to the two lines below. So the only difference is whether `video_stream` and `audio_stream` defined below should be containers or streams. Given the naming below it made more sense to me to make them NOT containers.

Reply:
I think either I'm missing something or you are :).

So what you are saying is FakePyAvStreams have no relation to an object in  PyAv

For example:
https://github.com/PyAV-Org/PyAV/blob/main/av/container/streams.pxd?

I assumed this was a fake version of above? If it is a fake version of above, it's fake version should match member types. Ie lists should be lists.

But i might be missing something.

Ps. I'm not saying the change is wrong. I'm just trying to understand if something was wrong (from my perspective) before.

Reply:
Yes, `FakePyAvStreams` have a relation to the `StreamContainer` PyAv object. But In both before the pull and after, `FakePyAVStreams.audio` and `FakePyAVStreams.video` are both containers of `FakePyAvStream`. I just changed where the containers are created - previously the `FakePyAvStreams` constructor was taking in containers of 1 stream to its parameters `video_stream` and `audio_stream`. Due to the parameter names, I thought it made more sense that those constructor parameters should simply each be a `FakePyAvStream`, and the constructor could take care of putting those into list containers.
Just look at these two lines and the two changed lines below in `MockPyAv`. The only difference is how `video_stream` and `audio_stream` are defined down there and how they are passed in to the constructor here. It doesn't really affect the `MockPyAv` class because that class is not related to a real PyAv class.